libs/filamat/src/MaterialBuilder.cpp

```diff
@@ -794,4 +694,45 @@ const std::string MaterialBuilder::peek(filament::backend::ShaderType type,
     return std::string("");
 }
 
+void MaterialBuilder::writeChunks(ChunkContainer& container, MaterialInfo& info) const noexcept {
+    EMIT_SIMPLE_CHUNK(uint32_t, ChunkType::MaterialVersion, filament::MATERIAL_VERSION);
+    EMIT_SIMPLE_CHUNK(const char*, ChunkType::MaterialName, mMaterialName.c_str_safe());
+    EMIT_SIMPLE_CHUNK(uint8_t, ChunkType::MaterialShading, static_cast<uint8_t>(mShading));
+    EMIT_SIMPLE_CHUNK(uint8_t, ChunkType::MaterialBlendingMode, static_cast<uint8_t>(mBlendingMode));
+
+    if (mBlendingMode == BlendingMode::MASKED) {
+        EMIT_SIMPLE_CHUNK(float, ChunkType::MaterialMaskThreshold, mMaskThreshold);
+    }
+
+    if (mShading == Shading::UNLIT) {
+        EMIT_SIMPLE_CHUNK(bool, ChunkType::MaterialShadowMultiplier, mShadowMultiplier);
+    }
+
+    EMIT_SIMPLE_CHUNK(bool, ChunkType::MaterialSpecularAntiAliasing, mSpecularAntiAliasing);
+    EMIT_SIMPLE_CHUNK(float, ChunkType::MaterialSpecularAntiAliasingVariance, mSpecularAntiAliasingVariance);
```

Comment:
This reminds me me that `MaterialSpecularAntiAliasingVariance` and `...Threshold` only needs to be written when `MaterialSpecularAntiAliasing` is true. The reading code would have to be updated to deal with this I believe. Not a big deal though, you can ignore it.

Reply:
Thanks for the heads up. Materials should have only the chunks they need so maybe I'll make that change in the future.